remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+#requires vcf2bed, bowtie2, samtools, bedtools and reverse_strand.py (uses bamnostic)
+
+#This basic pipeline allows you to input two assemblies and a variant vcf file, 
+#and remap the variants to the new genome
+
+
+#example command:
+#time ./remapping_commands.sh -g droso_dm3.fasta -n GCA_000001215.4_Release_6_plus_ISO1_MT_genomic.fna -a GCA_000001215.4 -v droso_variants_renamed.vcf -o test.vcf
+
+
+#To be added:
+# -support for indels
+# -find a way to extract new assembly accession so it's not required as input? or just use new assembly file name?
+# -decide what to do in the case of new REF allele being the same as the variant allele (was A > G on old assembly, now is G > G on new assembly)
+
+#Completed:
+# -filter only SNPs (not indels) using bcftools filter
+# -copy filter etc columns from original vcf
+# -copy vcf header to the new vcf file, but regenerate contigs from the list of contigs in new assembly, and change reference tag
+# -take into account reverse strand
+# -create a command line version where you input file names
+# -% of remapped variants calculation
+
+
+oldgenome=''
+newgenome=''
+newgenomeaccession=''
+vcffile=''
+outfile=''
+
+while getopts 'g:n:a:v:o:' flag; do
+	case "${flag}" in
+		g) oldgenome="${OPTARG}";;
+		n) newgenome="${OPTARG}" ;;
+		a) newgenomeaccession="${OPTARG}" ;;
+		v) vcffile="${OPTARG}" ;;
+		o) outfile="${OPTARG}" ;;
+	esac
+done
+
+
+#1) Flanking sequence generation
+echo "-----------------------1) Flanking sequence generation-----------------------"
+#generate chrom sizes:
+samtools faidx $oldgenome 
+cut -f1,2 "$oldgenome".fai > "$oldgenome".chrom.sizes
+
+#filter SNPs only
+bcftools filter -i 'TYPE="snp"' $vcffile > snps_only.vcf
+
+#store header
+awk '$1 ~ /^#/ {print $0}' snps_only.vcf > vcf_header.txt
```

Comment:
FYI this can be achieved more simply by `grep '^#'` :) However, the problem with the awk/grep approach is that they will scan the entire file, which can be millions of line long. It's better to use `bcftools view --header-only` to extract the header

Reply:
(It will only scan up until the `#CHROM POS ...` line and then quit)